Fix grammar test (#272)
flyteorg/flytectl: proposal/README.md

# Introduction

> This document proposes, **Flytectl** as one singular CLI that interacts with FlyteAdmin service. It is proposed to write the CLI in **Golang** and would support both gRPC and REST endpoints of

Reviewer: CLI or cli?

FlyteAdmin. We will start with gRPC endpoint as the client is easily generated and in future we should investigate generation of a Swagger based REST client from the gRPC specification. As we build
more SDK's in different languages we should support a common way of interacting with the API. This does no mean that some SDK's may provide native ways of interacting with the Admin API (for e.g.

> Flytekit), but the intention is that we will eventually replace **Flytekit/Flytecli** with Flytectl exclusively.

Reviewer: ```suggestion
Flytekit), but the intention is to eventually replace **Flytekit/Flytecli** with Flytectl exclusively.
```

> We also recommend that the design of Flytectl is careful and it could helps us with delivering user features faster without having to rely on the UI. Flytectl with follow standard oauth2 for

Reviewer: ```suggestion
We also recommend that the design of Flytectl is careful, and it could help us deliver user features faster without having to rely on the UI. Flytectl will follow standard oauth2 for
```

# Why Golang?
- Most of Flytebackend is written in golang
- Most of Flyte backend is written in Golang
- Golang offers great CLI tooling support with viper and cobra
- Golang toolchain to create cross-compiled small, light weight binary is really efficient and easy to use
- We already generate golang proto and clients for all our IDL
- we have multiple common libraries available to ease the development of this tool

> - kubectl is a stellar example of a cli done well

Reviewer: cli or CLI?

## Generating Swagger code

> We started exploring this (Flytetools)[https://github.com/lyft/flytetools#tools] has some work. We were also approached by the Swagger code gen maintainer to see if they can help.

Reviewer: ```suggestion
We started exploring this (Flytetools)[https://github.com/lyft/flytetools#tools] has some work. The Swagger code-gen maintainer also approached us to see if they could help.
```
